doc/release/1.7.3-notes.rst

```diff
@@ -0,0 +1,37 @@
+==========================
+SciPy 1.7.3 Release Notes
+==========================
+
+.. contents::
+
+SciPy 1.7.3 is a bug-fix release that provides binary wheels
+for MacOS arm64 with Python 3.8, 3.9, and 3.10. Note that a small
+number of test failures have been skipped for these new wheels.
```

Comment:
oh, I should likely mention the MacOS 12+ requirement

Reply:
Yes, that's kinda critical. It should reference gh-14688 for why we won't release 11.0 wheels